mm2src/lp_ordermatch.rs

```diff
@@ -3285,6 +3327,96 @@ pub async fn set_price(ctx: MmArc, req: Json) -> Result<Response<Vec<u8>>, Strin
+    let mut my_maker_orders = ordermatch_ctx.my_maker_orders.lock().await;
+
+    let my_order = match my_maker_orders.get(&req.uuid) {
+        Some(order) => order,
+        None => return ERR!("There is no order with UUID {}", &req.uuid),
+    };
+
+    let base = my_order.base.as_str();
+    let base_coin: MmCoinEnum = match try_s!(lp_coinfind(&ctx, base).await) {
+        Some(coin) => coin,
+        None => return ERR!("Base coin {} has been removed from config", base),
+    };
+
+    let rel = my_order.rel.as_str();
+    let rel_coin: MmCoinEnum = match try_s!(lp_coinfind(&ctx, rel).await) {
+        Some(coin) => coin,
+        None => return ERR!("Rel coin {} has been removed from config", rel),
+    };
+
+    let mut builder = MakerOrderBuilder::new(&base_coin, &rel_coin)
+        .with_max_base_vol(my_order.max_base_vol.clone())
+        .with_min_base_vol(Some(my_order.min_base_vol.clone()))
+        .with_price(my_order.price.clone());
+    let mut update_msg = new_protocol::MakerOrderUpdated::new(my_order.uuid);
+
+    if let Some(new_volume) = req.new_volume {
+        try_s!(
+            check_balance_for_maker_swap(
+                &ctx,
+                &base_coin,
+                &rel_coin,
+                new_volume.clone(),
+                None,
+                None,
+                FeeApproxStage::OrderIssue
+            )
+            .await
+        );
+        builder = builder.with_max_base_vol(new_volume.clone());
+        update_msg = update_msg.with_new_max_volume(new_volume.into());
+    };
+
+    if let Some(true) = req.max {
+        let max_volume = try_s!(get_max_volume(&ctx, &base_coin, &rel_coin).await);
+        builder = builder.with_max_base_vol(max_volume.clone());
+        update_msg = update_msg.with_new_max_volume(max_volume.into());
+    };
+
+    if let Some(min_volume) = req.min_volume {
+        builder = builder.with_min_base_vol(Some(min_volume.clone()));
+        update_msg = update_msg.with_new_min_volume(min_volume.into());
+    };
+
+    if let Some(new_price) = req.new_price {
+        builder = builder.with_price(new_price.clone());
+        update_msg = update_msg.with_new_price(new_price.into());
+    }
+
+    let conf_settings = OrderConfirmationsSettings {
+        base_confs: req
+            .base_confs
+            .unwrap_or_else(|| my_order.conf_settings.unwrap().base_confs),
+        base_nota: req
+            .base_nota
+            .unwrap_or_else(|| my_order.conf_settings.unwrap().base_nota),
+        rel_confs: req
+            .rel_confs
+            .unwrap_or_else(|| my_order.conf_settings.unwrap().rel_confs),
+        rel_nota: req.rel_nota.unwrap_or_else(|| my_order.conf_settings.unwrap().rel_nota),
+    };
+
+    builder = builder.with_conf_settings(conf_settings);
+    update_msg = update_msg.with_new_conf_settings(conf_settings);
+
+    maker_order_updated_p2p_notify(ctx.clone(), &my_order.base, &my_order.rel, update_msg).await;
```

Comment:
This should be called only after all validations pass and the order is really updated. In the current state, it's possible that we notify other nodes but the update won't really happen and cause orderbook synchronization issues.

Reply:
Should it be called right before the return statement after updating my_maker_orders entry? if so I think it should be fixed also in the `set_price` function here https://github.com/KomodoPlatform/atomicDEX-API/blob/01349d8e85b9be363bbad05a1ae342910bac9bfa/mm2src/lp_ordermatch.rs#L3281
as it should be after https://github.com/KomodoPlatform/atomicDEX-API/blob/01349d8e85b9be363bbad05a1ae342910bac9bfa/mm2src/lp_ordermatch.rs#L3284